SeaPublicWebsite/Controllers/EnergyEfficiencyController.cs

```diff
@@ -1233,6 +1233,7 @@ public async Task<IActionResult> AnswerSummary_Post(string reference)
                 ).ToList();
             }
 
+            propertyData.RecommendationsFirstRetrievedAt ??= DateTime.Now.ToUniversalTime();
```

Comment:
In that case @Glenn-Clarke I don't think it would be too hard to make a start on trying to make this more testable:

I think we could pull out lines 1217-1238 to a separate `EnergyEfficiencyService` or `PropertyDataService` without too much trouble. That would be easier to test in isolation, without having to worry about all the controller-level stuff.

(this is one of the big reasons why having big bloated controllers is something to avoid)